src/Optimizer/Transformer/RewriteAmpUrls.php

```diff
@@ -176,7 +176,7 @@ private function usesAmpCacheUrl($url)
             return false;
         }
 
-        return strpos($url, Amp::CACHE_HOST) === 0;
+        return substr_compare($url, Amp::CACHE_HOST, 0, strlen(Amp::CACHE_HOST)) === 0;
```

Comment:
I don't think this is an improvement. It is both more complex and slower, as I'd guess we never have 10MB URLs to begin with.

Reply:
@westonruter Are you okay with reverting this change?

Reply:
Sure, go ahead.